### PR TITLE
Update docker container for Pytorch 1.7.1

### PR DESCRIPTION
**Pull request by ChaoFang-TRI**
# Description

Update docker container for Pytorch 1.7.1
Create document for using docker
Fix small bug in existing docker file

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [x] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docker/WORKSPACE.cu111.docker [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    name = "googletest",
+    remote = "https://github.com/google/googletest",
+    commit = "703bd9caab50b139428cea1aaff9974ebee5742e",
+    shallow_since = "1570114335 -0400"
+)
+
+# CUDA should be installed on the system locally
+new_local_repository(
+    name = "cuda",
+    path = "/usr/local/cuda-11.1/",
```
Comment (narendasan) [inline docker/WORKSPACE.cu111.docker:38]: I think we can use a single WORKSPACE file for all dockerfiles if we make this `/usr/local/cuda`

File: docker/Dockerfile.20.10 [hunk 2]
```diff
@@ -0,0 +1,40 @@
+FROM nvcr.io/nvidia/pytorch:20.10-py3
+
+RUN apt-get update && apt-get install -y curl gnupg  && rm -rf /var/lib/apt/lists/*
+
+RUN curl https://bazel.build/bazel-release.pub.gpg | apt-key add - && \
+    echo "deb [arch=amd64] https://storage.googleapis.com/bazel-apt stable jdk1.8" | tee /etc/apt/sources.list.d/bazel.list
+
+RUN apt-get update && apt-get install -y bazel-3.4.1 && rm -rf /var/lib/apt/lists/*
```
Comment (narendasan) [inline docker/Dockerfile.20.10:8]: This should be Bazel 3.7.1

**Comment by narendasan** [comment]
Thanks for the PR, left a couple comments to make it more general

File: docker/WORKSPACE.cu111.docker (hunk 1 above)
Comment (ChaoFang-TRI) [reply docker/WORKSPACE.cu111.docker]: Done, nice suggestion!

File: docker/Dockerfile.20.10 (hunk 2 above)
Comment (ChaoFang-TRI) [reply docker/Dockerfile.20.10]: Updated

**Review by narendasan — APPROVED** [review]
Cool these changes look good to me! Thanks!

**Pull request merged**